internal/docker/containerfollowrun.go

```diff
@@ -26,7 +29,25 @@ func (c *Client) ContainerFollowRun(ctx context.Context, containerID string, for
 		dsterr = out.Default.ErrWriter()
 	}
 
-	_, err = stdcopy.StdCopy(dstout, dsterr, buf.Reader)
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, os.Interrupt)
 
-	return err
+	done := make(chan struct{})
+
+	go func() {
+		_, _ = stdcopy.StdCopy(dstout, dsterr, buf.Reader)
+		close(done)
+	}()
+
+	select {
+	case <-sigCh:
+		stopCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		_ = c.ContainerStop(stopCtx, containerID, container.StopOptions{})
+
+		<-done
+	case <-done:
+	}
+
+	return nil
```

Comment:
This part is a specific logic, and I would suggest keeping it in a separate function with its own, self-describing name like `stopContainerOnSigInt`.